host/class/uvc/usb_host_uvc_2/include/usb/uvc_host.h

```diff
@@ -0,0 +1,241 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <stdbool.h>
+#include "usb/usb_host.h"
+#include "usb/usb_types_uvc.h"
+#include "esp_err.h"
+
+// Use this macros for opening a UVC stream with any VID or PID
+#define UVC_HOST_ANY_VID (0)
+#define UVC_HOST_ANY_PID (0)
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct uvc_host_stream_s *uvc_host_stream_hdl_t;
+
+/**
+ * @brief Configuration structure of USB Host UVC driver
+ */
+typedef struct {
+    size_t driver_task_stack_size; /**< Stack size of the driver's task */
+    unsigned driver_task_priority; /**< Priority of the driver's task */
+    int xCoreID;                   /**< Core affinity of the driver's task */
+    bool create_background_task;   /**< When set to true, background task handling usb events is created.
+                                        Otherwise user has to periodically call uvc_host_handle_events function */
+} uvc_host_driver_config_t;
+
+/**
+ * @brief Event types of this driver
+ */
+enum uvc_host_dev_event {
+    UVC_HOST_ERROR,                  /**< USB transfer error */
+    UVC_HOST_DEVICE_DISCONNECTED,    /**< Device was suddenly disconnected. The stream is stopped. */
+    UVC_HOST_FRAME_BUFFER_OVERFLOW,  /**< The received frame was discarded because it exceeded the available frame buffer space.
```

Comment:
Is that also an error? Or it is possible to increase the `frame_size` right after getting this event? 

If this is an error, maybe it could be better: 
```suggestion
    UVC_HOST_ERROR_FRAME_BUFFER_OVERFLOW,  /**< The received frame was discarded because it exceeded the available frame buffer space.
```

Reply:
The frame buffer size cannot be changed of the UVC stream was opened (I can add this into TODO list)

It's not a real error because the stream continues. It just informs the user that a frame was dropped. If this happens too often, the user can closed the stream and open it again with larger frame buffers...

Reply:
Then maybe? 

```suggestion
    UVC_HOST_DROPPED_FRAMES,  /**< The received frame was discarded because it exceeded the available frame buffer space.
```